### PR TITLE
CAS-375 CAS-377 CAS-376 Implement attachments dialog

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://stream-io.atlassian.net/browse/CAS-376
https://stream-io.atlassian.net/browse/CAS-375
https://stream-io.atlassian.net/browse/CAS-377

### Description

Implemented attachment dialog which allows to:

- pick images/videos from the gallery
- pick recent files
- pick files from a file manager
- take photo / record video

### Checklist

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [X] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [X] Comparison screenshots added for visual changes
- [X] Reviewers added

### CODE REVIEW

**Review by ogkuzmin — COMMENTED** [review]
There should be some tests for check logic, shouldn't be? Especially for logic to handle attachment selection state in adapters

**Comment by bychkovdmitry** [comment]
> There should be some tests for check logic, shouldn't be? Especially for logic to handle attachment selection state in adapters

Agree, will try to add them. Hope the PR will not be too big.

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/attachments/media/MediaAttachmentFragment.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        binding.grantPermissionsInclude.apply {
+            grantPermissionsImageView.setImageResource(R.drawable.stream_attachment_permission_media)
+            grantPermissionsTextView.setText(R.string.stream_attachment_dialog_permission_media)
+            grantPermissionsTextView.setOnClickListener {
+                tryPopulateAttachments()
+            }
+        }
+    }
+
+    private fun tryPopulateAttachments() {
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/attachments/media/MediaAttachmentFragment.kt:91]: Seems it should be called `checkPermissions`

File: stream-chat-android-ui-components/src/main/res/layout/stream_fragment_attachment_camera.xml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/res/layout/stream_fragment_attachment_camera.xml:2]: Why not just simple FrameLayout? Is constraint required here?

**Review by ogkuzmin — APPROVED** [review]
LGTM. Just add tests later)

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/attachments/media/MediaAttachmentFragment.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/attachments/media/MediaAttachmentFragment.kt:91]: Changed corresponding method names in `CameraAttachmentFragment`, `FileAttachmentFragment`, `MediaAttachmentFragment`

File: stream-chat-android-ui-components/src/main/res/layout/stream_fragment_attachment_camera.xml (hunk 2 above)
Comment (bychkovdmitry) [reply stream-chat-android-ui-components/src/main/res/layout/stream_fragment_attachment_camera.xml:2]: Lets leave it as is for now, `stream_attachment_permission_view.xml` include is used in different fragment layouts. And the included layout relies on the fact that it is included into `ConstraintLayout`.

**Comment by bychkovdmitry** [comment]
Create a Jira task to cover the functionality with tests

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**